.github/workflows/ci.yml

```diff
@@ -62,18 +61,24 @@ jobs:
                   git clone https://github.com/vivid-planet/comet-admin-lang.git
                   cp -r lang/* comet-admin-lang/
                   cd comet-admin-lang
-                  git config user.email "vivid-planet-bot@vivid-planet.com"
                   git add .
                   if [[ `git status --porcelain` ]]; then git commit -m "add translatable strings for ${{ github.sha }}" && git remote rm origin && git remote add origin https://vivid-planet-bot:${{ secrets.GH_TOKEN }}@github.com/vivid-planet/comet-admin-lang.git && git push --set-upstream origin master; fi
 
-            - name: Publish Canary release
-              if: ${{ github.ref_name == 'main' || github.ref_name == 'next' }}
-              run: yarn publish:release --canary --preid=canary.${{ github.run_number }} --dist-tag=canary --yes
+            - name: Publish premajor canary release
+              if: |
+                  github.event.label.name != 'skip-release' &&
+                  github.base_ref == 'next' &&
+                  github.event.pull_request.merged == true
+              run: yarn publish:release premajor --no-push --no-git-tag-version --canary --preid=canary.${{ github.run_number }} --dist-tag=canary --yes
               env:
                   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
-            - name: Publish stable release
-              if: ${{ github.ref_type == 'tag' }}
-              run: yarn publish:release ${{ github.ref_name }} --yes
+            - name: Publish stable minor release
+              if: |
+                  github.event.label.name != 'skip-release' &&
+                  github.base_ref == 'main' &&
+                  github.event.pull_request.merged == true
+              run: yarn publish:release minor --yes
               env:
                   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+                  GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```

Comment:
this will commit now the changes into main. Can only be tested when next comes into main